src/runtime.js

```diff
@@ -209,6 +209,21 @@ function adoptShared(global, callback) {
   // locking).
   for (let i = 0; i < preregisteredExtensions.length; i++) {
     const fnOrStruct = preregisteredExtensions[i];
+    if (IS_ESM) {
```

Comment:
Worth verifying (if only for the future) that if/else conditions are DCE'd for replaced values.

Reply:
yep, i've confirmed that this gets DCE while the old `getMode().esm` one doesnt